src/Utilities/Math.hpp

```diff
@@ -79,3 +79,24 @@ template <typename T, Requires<std::is_arithmetic<T>::value> = nullptr>
 auto invcbrt(const T& arg) noexcept {
   return static_cast<T>(1.0) / cbrt(arg);
 }
+
+namespace sgn_detail {
+template <typename T>
+constexpr int sgn(const T& val, std::true_type /*is_signed*/) noexcept {
+  return static_cast<int>(static_cast<bool>(static_cast<T>(0) < val)) -
```

Comment:
probably doesn't work with DataVectors? If I recall correctly, our blaze versions don't support comparison operation between vectors and scalars, and the casts won't do sensible things.
Not that I have a pressing use-case for vectors, just that we'll need to add template specializations if we want it later.

Reply:
Yea, I doubt it'll work with Blaze vectors, but like you say, it's also not obvious if we'll ever need that support, and if so we probably want to do something vectorized anyway.